WeScan/Scan/ScannerViewController.swift

```diff
@@ -137,46 +143,94 @@ final class ScannerViewController: UIViewController {
-        } else {
-            cancelButtonBottomConstraint = view.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
-            shutterButtonBottomConstraint = view.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
-        }
-        
-        let cancelButtonConstraints = [
-            cancelButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 24.0),
-            cancelButtonBottomConstraint
-            ]
-        
-        let shutterButtonConstraints = [
+        shutterButtonConstraints = [
             shutterButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            shutterButtonBottomConstraint,
             shutterButton.widthAnchor.constraint(equalToConstant: 65.0),
             shutterButton.heightAnchor.constraint(equalToConstant: 65.0)
-            ]
+        ]
         
-        let activityIndicatorConstraints = [
+        activityIndicatorConstraints = [
             activityIndicator.centerXAnchor.constraint(equalTo: view.centerXAnchor),
             activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
         ]
         
-        NSLayoutConstraint.activate(quadViewConstraints + cancelButtonConstraints + shutterButtonConstraints + activityIndicatorConstraints)
+        if #available(iOS 11.0, *) {
+            let window = UIApplication.shared.keyWindow
+            
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
+            ]
+            
+            if let safeAreaInsets = window?.safeAreaInsets {
+                toolbarConstraints.append(toolbar.heightAnchor.constraint(equalToConstant: safeAreaInsets.top + 44.0))
+            }
+            
+            cancelButtonConstraints = [
+                cancelButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24.0),
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+            ]
+            
+            let shutterButtonBottomConstraint = view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
+            shutterButtonConstraints.append(shutterButtonBottomConstraint)
+        } else {
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
```

Comment:
As far as I know it shouldn't be needed to use autolayout for the UIToolBar if used correctly. The OS should be able to manager it's position, right @Boris-Em?

If so, I would love to fix it in that way rather than writing autolayout logic for it.

Reply:
Hey, the OS/navigation controller's toolbar is indeed managed, but it's also fixed, preventing us from putting it at the top of the screen on the scanner view (however, in #80 i'm using the provided toolbar for the 'review' screen). I was previously using frame's here but autolayout makes sure the toolbar will work on future updates to iOS